app/Http/Controllers/Admin/UserController.php

```diff
@@ -44,16 +44,28 @@ public function index()
      */
     public function store(Request $request)
     {
-        $request->validateWithBag('createUser', [
-            'name' => ['required', 'string', self::MAX_255],
-            'email' => ['required', 'string', 'email', self::MAX_255, 'unique:users'],
-            'password' => ['required', 'string', self::MAX_255, 'confirmed'],
-        ]);
+        // Randomly creates a temporary password when it is not set.
+        // The user can login using a reset link.
+        $password = $request->password;
+        if (!$password) {
+            $password = Str::random(20);
+
+            $request->validateWithBag('createUser', [
+                'name' => ['required', 'string', self::MAX_255],
+                'email' => ['required', 'string', 'email', self::MAX_255, 'unique:users']
+            ]);
+        } else {
+            $request->validateWithBag('createUser', [
+                'name' => ['required', 'string', self::MAX_255],
+                'email' => ['required', 'string', 'email', self::MAX_255, 'unique:users'],
+                'password' => ['required', 'string', self::MAX_255, 'confirmed', 'min:8'],
+            ]);
+        }
```

Comment:
```suggestion
        $request->validateWithBag('createUser', [
            'name' => ['required', 'string', self::MAX_255],
            'email' => ['required', 'string', 'email', self::MAX_255, 'unique:users'],
            'password' => ['string', self::MAX_255, 'confirmed', 'min:8'],
        ]);
        
        // Randomly creates a temporary password when it is not set.
        // The user can login using a reset link.
        $password = $request->password;
        if (!$password) {
            $password = Str::random(20);
        }
```
no need to duplicate validation when you can remove the require rule

Reply:
Does the 'min:8' rule not apply when there is no 'required' rule? Does it only apply when there is at least a character?

Reply:
It still applies, it also applies on 'string', this logic is not the right behavior, I will revert this change.

Reply:
odd but this isn't a super important change. let's merge